packages/app/src/explorer/NxBadge.tsx

```diff
@@ -0,0 +1,31 @@
+import type { Entity } from '@h5web/shared';
+import { isGroup } from '@h5web/shared';
+import { useContext } from 'react';
+
+import { ProviderContext } from '../providers/context';
+import styles from './Explorer.module.css';
+import { needsNxBadge } from './utils';
+
+interface Props {
+  entity: Entity;
+}
+
+function NxBadge(props: Props) {
+  const { entity } = props;
+  const { attrValuesStore } = useContext(ProviderContext);
+
+  if (!isGroup(entity) || !needsNxBadge(entity, attrValuesStore)) {
```

Comment:
The fact that non-group entities will never get a NX badge is a side-effect from the NX spec. As a consequence, it would make more sense to me to check for `isGroup` inside `needsNxBadge` (which would accept an entity)